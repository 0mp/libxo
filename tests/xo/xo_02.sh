#
# $Id$
#
# Copyright 2019, Juniper Networks, Inc.
# All rights reserved.
# This SOFTWARE is licensed under the LICENSE provided in the
# ../Copyright file. By downloading, installing, copying, or otherwise
# using the SOFTWARE, you agree to be bound by the terms of that
# LICENSE.

XO=$1
shift

XOP="${XO} --warn"

# This is testing --wrap, --open, --close, --top-wrap, etc, so
# the output is not a single valid document

set -- 'The capital of {:state} is {:city}\n' 'North Carolina' Raleigh

${XOP} --top-wrap --open a/b/c "$@"
${XOP} --top-wrap --close a/b/c --not-first "$@"

${XOP} --top-wrap --wrap a/b/c "$@"

${XOP} --top-wrap --open a/b/c "$@"
${XOP} --depth 4 --not-first --wrap d/e/f "$@"
${XOP} --top-wrap --close a/b/c --not-first "$@"

${XOP} --wrap a/b/c "$@"

${XOP} --top-wrap --wrap a/b/c "$@"

${XOP} --top-wrap "test\n"

${XOP} --open answer
${XOP} "Answer:"
${XOP} --continuation "$@"
${XOP} --close answer

${XOP} --help
