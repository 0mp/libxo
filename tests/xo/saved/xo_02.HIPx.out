<div class="line">
  <div class="text">The capital of </div>
  <div class="data" data-tag="state" data-xpath="/a/b/c/state">North Carolina</div>
  <div class="text"> is </div>
  <div class="data" data-tag="city" data-xpath="/a/b/c/city">Raleigh</div>
</div>
<div class="line">
  <div class="text">The capital of </div>
  <div class="data" data-tag="state" data-xpath="/state">North Carolina</div>
  <div class="text"> is </div>
  <div class="data" data-tag="city" data-xpath="/city">Raleigh</div>
</div>
<div class="line">
  <div class="text">The capital of </div>
  <div class="data" data-tag="state" data-xpath="/a/b/c/state">North Carolina</div>
  <div class="text"> is </div>
  <div class="data" data-tag="city" data-xpath="/a/b/c/city">Raleigh</div>
</div>
<div class="line">
  <div class="text">The capital of </div>
  <div class="data" data-tag="state" data-xpath="/a/b/c/state">North Carolina</div>
  <div class="text"> is </div>
  <div class="data" data-tag="city" data-xpath="/a/b/c/city">Raleigh</div>
</div>
<div class="line">
  <div class="text">The capital of </div>
  <div class="data" data-tag="state" data-xpath="/d/e/f/state">North Carolina</div>
  <div class="text"> is </div>
  <div class="data" data-tag="city" data-xpath="/d/e/f/city">Raleigh</div>
</div>
<div class="line">
  <div class="text">The capital of </div>
  <div class="data" data-tag="state" data-xpath="/state">North Carolina</div>
  <div class="text"> is </div>
  <div class="data" data-tag="city" data-xpath="/city">Raleigh</div>
</div>
<div class="line">
  <div class="text">The capital of </div>
  <div class="data" data-tag="state" data-xpath="/a/b/c/state">North Carolina</div>
  <div class="text"> is </div>
  <div class="data" data-tag="city" data-xpath="/a/b/c/city">Raleigh</div>
</div>
<div class="line">
  <div class="text">The capital of </div>
  <div class="data" data-tag="state" data-xpath="/a/b/c/state">North Carolina</div>
  <div class="text"> is </div>
  <div class="data" data-tag="city" data-xpath="/a/b/c/city">Raleigh</div>
</div>
<div class="line">
  <div class="text">test</div>
</div>
<div class="line">
  <div class="text">Answer:</div>
  <div class="text">The capital of </div>
  <div class="data" data-tag="state" data-xpath="/state">North Carolina</div>
  <div class="text"> is </div>
  <div class="data" data-tag="city" data-xpath="/city">Raleigh</div>
</div>
